src/pkg/term/colorizer.go

```diff
@@ -121,17 +121,45 @@ func output(w *termenv.Output, c Color, msg string) (int, error) {
 	if len(msg) == 0 {
 		return 0, nil
 	}
+	var buf strings.Builder
 	if doColor(w) {
-		w.WriteString(termenv.CSI + c.Sequence(false) + "m")
-		defer w.Reset()
+		Append(&buf, doColor(w), c, msg)
+		msg = buf.String()
 	}
 	return w.WriteString(msg)
 }
 
-func outputf(w *termenv.Output, c Color, format string, v ...any) (int, error) {
-	line := fmt.Sprintf(format, v...)
-	line = ensureNewline(line)
-	return output(w, c, line)
+const ResetColorStr = termenv.CSI + termenv.ResetSeq + "m"
+
+func Append(w io.Writer, canColor bool, c Color, v ...any) (int, error) {
+
+	var l int
+	if canColor {
+		n, err := io.WriteString(w, termenv.CSI+c.Sequence(false)+"m")
+		l += n
+		if err != nil {
+			return l, err
+		}
+	}
+	for _, s := range v {
+		n, err := fmt.Fprint(w, s)
+		l += n
+		if err != nil {
+			return l, err
+		}
+	}
+	if canColor {
+		n, err := io.WriteString(w, ResetColorStr)
```

Comment:
Do you need to make this block a defer higher up in case the for-loop above exits out of this function early?

Reply:
Good call!